src/games/planetwars/visualizer/controls.js

```diff
@@ -38,46 +38,30 @@ class Controls {
     });
   }
 
-  readLog(e) {
-    if (this.visualizer) {
-      this.hidePauseButton();
-      this.visualizer.clear();
-    }
-
-    var reader = new FileReader();
-    reader.onload = event => {
-      var log = event.target.result;
-      this.visualizer.visualize(log);
-      this.attachEvents(this.visualizer.turn_controller);
-    };
-
-    reader.readAsText(e.target.files[0]);
-  }
-
-  attachEvents(turn_controller) {
+  attachEvents(model) {
     d3.select('#hide_score').attr("hidden", null);
 
     d3.select('#play').on("click", e => {
-      turn_controller.play();
+      model.run_binder.update(true);
     });
 
     d3.select('#pause').on("click", e => {
-      turn_controller.pause();
+      model.run_binder.update(false);
     });
 
     d3.select('#next').on("click", e => {
-      turn_controller.nextTurn();
+      model.turn_binder.update((model.turn_binder.value) + 1);
     });
 
     d3.select('#previous').on("click", e => {
-      turn_controller.previousTurn();
+      model.turn_binder.update((model.turn_binder.value) - 1);
     });
 
     d3.select('#speeddown').on("click", e => {

```

Comment:
Speeddown an speedup need a seperate function. Maybe do this consistently for all callbacks.